metadata_backend/api/middlewares.py

```diff
@@ -1,53 +1,113 @@
+        elif error.status == 401:
+            raise web.HTTPUnauthorized(text=details,
+                                       content_type=c_type)
+        elif error.status == 403:
+            raise web.HTTPForbidden(text=details,
+                                    content_type=c_type)
+        elif error.status == 404:
+            raise web.HTTPNotFound(text=details,
+                                   content_type=c_type)
+        else:
+            raise web.HTTPServerError()
+
+    return http_error_handler
 
-    @middleware
-    async def http_error_handler(req: Request, handler: Callable) -> Response:
-        """Handle HTTP errors.
 
-        :param req: A request instance
-        :param handler: A request handler
-        :raises: Reformatted HTTP Exceptions
-        :returns: Successful requests unaffected
-        """
+@middleware
+async def jwt_authentication(req: Request, handler: Callable) -> Response:
+    """Middleware for validating and authenticating JSON web token.
+
+    :param req: A request instance
+    :param handler: A request handler
+    :raises: HTTP Exception with status code 401 or 403
+    :returns: Successful requests unaffected
+    """
+    if 'Authorization' in req.headers:
+        # Check token exists
         try:
-            response = await handler(req)
-            return response
-        except web.HTTPError as error:
-            details = _json_exception(error.status, error, req.url)
-            LOG.error(details)
-            c_type = 'application/problem+json'
-            if error.status == 400:
-                raise web.HTTPBadRequest(text=details,
-                                         content_type=c_type)
-            elif error.status == 401:
-                raise web.HTTPUnauthorized(text=details,
-                                           content_type=c_type)
-            elif error.status == 403:
-                raise web.HTTPForbidden(text=details,
-                                        content_type=c_type)
-            elif error.status == 404:
-                raise web.HTTPNotFound(text=details,
-                                       content_type=c_type)
-            else:
-                raise web.HTTPServerError()
+            scheme, token = req.headers.get('Authorization').split(' ')
+            LOG.info('Auth token received.')
+        except ValueError as err:
+            raise web.HTTPUnauthorized(reason=f"Failure to read token: {err}")
 
-    return http_error_handler
+        # Check token has proper scheme and was provided.
+        if not re.match('Bearer', scheme):
+            raise web.HTTPUnauthorized(reason="Invalid token scheme, "
+                                              "Bearer required.")
+        if token is None:
+            raise web.HTTPUnauthorized(reason='Token cannot be empty.')
+
+        # JWK and JWTClaims parameters for decoding
+        key = environ.get('PUBLIC_KEY', None)
```

Comment:
we resolve to getting a public key locally when we cannot reach a JWK endpoint, but usually we use a JWK endpoint to get the public key.